src/WrappedHttpHandler.php

```diff
@@ -66,18 +72,41 @@ public function __invoke(
         RequestInterface $request
     ) {
         $fn = $this->httpHandler;
+        $options = $command['@http'] ?: [];
+        $statsPromise = $this->getStatsPromise($options);
 
-        return Promise\promise_for($fn($request, $command['@http'] ?: []))
+        return Promise\promise_for($fn($request, $options))
             ->then(
-                function (ResponseInterface $res) use ($command, $request) {
-                    return $this->parseResponse($command, $request, $res);
+                function (ResponseInterface $response) use (
+                    $command,
+                    $request,
+                    $statsPromise
+                ) {
+                    return $statsPromise->then(function (array $stats) use (
```

Comment:
Given that this class is used to power almost every transaction, it might be better to only use the additional layer of promise if the user requires transfer stats.

I also don't know of a case where you would get the `then` callback invoked and the on_stats callback would not have been completed. I wonder if there's a different way to get at this data without a promise.